src/services/TodoService.ts

```diff
@@ -1,63 +1,38 @@
-      },
-      // TODO: if a component would like to implement a different behavior of error handling, they won't be able to because you pop up a swal every error,
-      // that's not a good design call.
-      // I know that you might have done it because it served your needs in this small app, but I wanted you 
-      // to write the application as scaleable application
-      function (error: Error) {
-        Swal.fire({
-          title: "Something wrong happened",
-          text: error.message + ", The action you just did wasn't saved",
-          icon: "error",
-        });
-
-        return Promise.reject(error);
-      }
-    );
-  }
-
-  // TODO: no need to call the param `todoToSave` and or `saveTodo` method name,
-  // we're already in save method in TodoService class.
-  // keep in mind that naming conventions are important, don't add words that do not make the code clearer
-  // I'll repeat this comment in a few more places but not all of them only to make sure you understand 
-  // my intention
-  public static async saveTodo(todoToSave: Todo): Promise<Todo> {
-    const { data } = await axios.post(`/todos`, todoToSave);
+// ANSWER: I read, and I what I understood is that naming the imported entity everytime causes some problems, like refactoring
+// or auto complete by the IDE. Another problem is exporting big objects instead of dividing to smaller exports.
 
-    return data;
+export default class TodoService {
+  public static async save(
+    todo: Omit<Todo, "id">
+  ): Promise<AxiosResponse<Todo>> {
+    return axiosInstance.post(`/todos`, todo);
   }
 
   // TODO: please notice that delete of multiple todos at once is not supported.
   // no need to implement as long as you're aware of the problem with the UX.
-  // TODO: name the method `delete`, and the param `id` because it's obvious you want to delete a todo
-  public static async deleteTodo(todoToDeleteID: number): Promise<void> {
-    await axios.delete(`/todos/${todoToDeleteID}`);
+  // ANSWER: yeah i'm aware of this, json-server doesn't support the deletion of multiple rows
+  // so I didn't want to put the logic of deleting multiple todos in here so I implemented it in the component because I only use it once.
+  public static async delete(id: number): Promise<void> {
+    await axiosInstance.delete(`/todos/${id}`);
   }
 
-  // TODO: there's no need to use await in this case, anyway you return a promise and the user uses await or .then
   // TODO: please use `noImplicitAny` in tsconfig.json file and look at the errors. in this case you don't use the type safety TS provides.
-  public static async getTodos(): Promise<Array<Todo>> {
-    // TODO: no need to use `` format if you don't use params, make your conventions consistent all across the application.
-    const { data } = await axios.get(`/todos`);
-
-    return data;
+  // ANSWER: I added it to tsconfig, but it didn't put an error up on the function, I don't fully understand why it didn't throw an error
```

Comment:
maybe it does not throw an error here because if you look inside axios source code, their default is any, and ts might consider it as explicit any.

just make sure nothing else in your code doesn't throw any other error.
make sure you use `restart TS server` after you update it  (even though it watches the tsconfig file)